statsmodels/robust/scale.py

```diff
@@ -45,8 +45,11 @@ def mad(a, c=Gaussian.ppf(3/4.), axis=0, center=np.median):
     """
     a = array_like(a, 'a', ndim=None)
     c = float_like(c, 'c')
-    if callable(center) and a.size:
-        center = np.apply_over_axes(center, a, axis)
+    if a.size:
+        if callable(center):
+            center = np.apply_over_axes(center, a, axis)
+        else:
+            center = float_like(center, 'center')
     else:
         center = 0.0
```

Comment:
What case does this handle?
AFAICS, not a.size would only check for empty arrays.


```
np.median(np.abs(np.array([]) - 0))
nan
```

Reply:
I think you just need an `elif center is None` and then set center to 0, else cast check to float.

Reply:
If a is empty and I don't set center to something then the final line throws a ZeroDivisionError when running line 76 of test_scale.py, exactly the line that tests that mad returns nan when a is empty.